src/SprykerEco/Zed/Unzer/Business/Checkout/Mapper/UnzerCheckoutPostSaveMapper.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Unzer\Business\Checkout\Mapper;
+
+use ArrayObject;
+use Generated\Shared\Transfer\ItemTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use Generated\Shared\Transfer\UnzerBasketItemTransfer;
+use Generated\Shared\Transfer\UnzerBasketTransfer;
+use Generated\Shared\Transfer\UnzerPaymentResourceTransfer;
+use SprykerEco\Zed\Unzer\UnzerConfig;
+
+class UnzerCheckoutPostSaveMapper implements UnzerCheckoutMapperInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Unzer\UnzerConfig
+     */
+    protected $unzerConfig;
+
+    /**
+     * @param \SprykerEco\Zed\Unzer\UnzerConfig $unzerConfig
+     */
+    public function __construct(UnzerConfig $unzerConfig)
+    {
+        $this->unzerConfig = $unzerConfig;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\UnzerBasketTransfer $unzerBasketTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerBasketTransfer
+     */
+    public function mapQuoteTransferToUnzerBasketTransfer(
+        QuoteTransfer $quoteTransfer,
+        UnzerBasketTransfer $unzerBasketTransfer
+    ): UnzerBasketTransfer {
+        return $unzerBasketTransfer
+            ->setAmountTotalGross($quoteTransfer->getTotals()->getGrandTotal() / 100)
+            ->setAmountTotalVat($quoteTransfer->getTotals()->getTaxTotal()->getAmount() / 100)
```

Comment:
```suggestion
            ->setAmountTotalVat($quoteTransfer->getTotalsOrFail()->getTaxTotalOrFail()->getAmount() / 100)
```
same for similar cases when you use nested transfer properties